### ISSUE TITLE
Overlay flashes if in controlled mode, and next is outside the regular flow

### ISSUE THREAD

**Issue by scottwarren** [gilbarbara/react-joyride#918]
## To Reproduce

In controlled mode, have a button outside the default tooltip that has a delay before going to the next step

## Expected behavior

> A clear and concise description of what you expected to happen.

Overlay should not flicker

## Link to repl or repo (highly encouraged)

Please provide a https://codesandbox.io/ demo or similar.

https://codesandbox.io/s/react-joyride-overlay-flickering-bug-forked-mh25rm?file=/src/App.js

Issues without a reproduction link are likely to stall.

## Run `npx envinfo --system --binaries --npmPackages react-joyride`

Paste the results here:

```bash

  System:
    OS: macOS 13.3.1
    CPU: (8) arm64 Apple M1
    Memory: 79.48 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 16.14.0 - ~/.nvm/versions/node/v16.14.0/bin/node
    npm: 8.3.1 - ~/.nvm/versions/node/v16.14.0/bin/npm
  npmPackages:
    react-joyride: 2.5.4 => 2.5.4
```

**Comment by scottwarren** [comment]
As a note, this app isn't using `React.StrictMode`

**Comment by iamFoxx** [comment]
Same

**Comment by Ivan-dev-RED** [comment]
The same problem,,,

**Comment by Ivan-dev-RED** [comment]
@gilbarbara 
Do you have any ideas?
May be we need to install some package?

**Comment by Ivan-dev-RED** [comment]
This is fucking bullshit...
On `build` it works correctly.
Why does it flicking in development mode? I wasted some hours of my life on this crap...

**Comment by gilbarbara** [comment]
@Ivan-dev-RED 

There's no need to be aggressive. If you don't like this library, just create your own.

**Comment by iamFoxx** [comment]
@Ivan-dev-RED solve this by removing `React.StrictMode`. No need to be rude, that's not ok.

**Comment by juniorforlife** [comment]
@gilbarbara Unfortunately I’m running into this issue even in production (without StrictMode). I made an interactive tutorial by removing the Next button and adding setTimeout. After user clicks a spotlight, the next tooltip will be shown after a certain time. It works fine except for the overlay flickers. How do I fix this?

**Comment by juniorforlife** [comment]
@scottwarren have you found the fix?

**Comment by scottwarren** [comment]
> @scottwarren have you found the fix?

No, although I did manage to figure out a work around, I implemented my own overlay that I control which is in sync with which step is currently showing

**Comment by juniorforlife** [comment]
@scottwarren do you mind sharing a simple version of your customized overlay 🙏

**Comment by juniorforlife** [comment]
@gilbarbara Have you had a chance to look at this issue?

**Comment by gilbarbara** [comment]
Hey @juniorforlife 
Nope, and I won't be able to work with this for the foreseeable future.

**Comment by Adelao4** [comment]
Please share integration implementation

**Comment by gilbarbara** [comment]
I've refactored how the Overlay is rendered to prevent any flickering.
It's available in v2.8.0
